crates/astria-sequencer/src/transaction/mod.rs

```diff
@@ -72,15 +75,20 @@ pub(crate) async fn check_balance_mempool<S: StateReadExt + 'static>(
     Ok(())
 }
 
-// Checks that the account has enough balance to cover the total fees for all actions in the
-// transaction.
+// Checks that the account has enough balance to cover the total fees and transferred values
+// for all actions in the transaction.
+//
+// allow too many lines because of the number of actions and the need to check each one.
+#[allow(clippy::too_many_lines)]
```

Comment:
IMO this is a strong hint to have a bunch of free-standing functions for each of the actions, and delegate the checks there. I.e:

```rust
match action {
    Action::Transfer(act) => process_transfer_action(act, fees_by_asset)
    ...
}
```